### ISSUE TITLE
RStudio does not recognize installed R 

### ISSUE THREAD

**Issue by aolykhin** [rstudio/rstudio#14253]
### System details

    RStudio Edition : Desktop (Electron)
    RStudio Version : 2023.12.0+369
    OS Version      :  CentOS 7
    R Version       :  4.3.2

### Steps to reproduce the problem
Install R 4.3.2 with gcc compiler 10.2.0 in a non-standard directory, in my case /software/R-4.3.2-el7-x86_64/. Copy precompiled RStudio 2023.12 from the official website and set root ownership on chrome-sandbox and 4755 permissions. I load R 4.3.2 as a module, which can be replicated by pre-pending path variables:
export PKG_CONFIG_PATH=/software/R-4.3.2-el7-x86_64/lib64/pkgconfig/:$PKG_CONFIG_PATH
export LIBRARY_PATH=/software/R-4.3.2-el7-x86_64/lib64/R/lib:$LIBRARY_PATH
export LD_LIBRARY_PATH=/software/R-4.3.2-el7-x86_64/lib64:$LD_LIBRARY_PATH
export PATH=/software/R-4.3.2-el7-x86_64/bin/:$PATH

### Describe the problem in detail
I have installed the R version 4.3.2 and it works for me when using command line interface. I can load R shell and install R packages. Once R 4.3.2 module is loaded, I run RStudio binary and get the error that: "R doesn't appear to be installed. Please install R before running RStudio. 

### Describe the behavior you expected
I would expect the RStudio 2023.12 to work with R 4.3.2 because they are both latest versions and because RStudio 2023.12 works with the old version of R 3.5.2 installed on my system. Yes, it complains but starts the RStudio anyway with the following output in the terminal: [124547:0212/155325.520998:ERROR:gl_factory.cc(120)] Requested GL implementation (gl=none,angle=none) not found in allowed implementations: [(gl=egl-angle,angle=default),(gl=egl-gles2,angle=none)].

- [x ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by aolykhin** [comment]
Comment: I don't have root privileges to make a diagnostic report. I used RStudio Fedora 19/Red Hat 7 distribution; Currently 2023.12.1 is available but I don't think it will resolve the issue https://download1.rstudio.org/electron/centos7/x86_64/rstudio-2023.12.1-402-x86_64-fedora.tar.gz

**Comment by ronblum** [comment]
@aolykhin Thank you for raising this! When using R 3.5.2, did you install and configure it in the same way?

**Comment by aolykhin** [comment]
No, R-3.5.2 was compiled with the default gcc compiler 4.8.5.

It is possible that the gcc 10.2 compiler I used may have some issues. Instead, I installed R-4.3.1 with gcc 10.1.0 and openblas-0.3.26 but had the same error that rstudio would not determine the path to R. Luckily, there is a workaround to start rstudio by updating a path explicitly to the R wrapper in /software/R-4.3.1-el7-x86_64/bin/R : 

export LD_LIBRARY_PATH=/software/gcc-10.1.0-el7-x86_64/lib64:/software/openblas-0.3.26-el7-x86_64/lib:$LD_LIBRARY_PA
TH

It is interesting that the same version of rstudio can recognize R-3.5.2, which was compiled with the default gcc in /usr/bin/gcc.
